core/src/worker/data-feed.ts

```diff
@@ -385,6 +392,14 @@ function submitHeartbeatJob(heartbeatQueue: Queue, state: State, _logger: Logger
       (job) => job.opts.jobId == jobId
     )
 
+    if (!state.isActive({ oracleAddress })) {
+      logger.warn(
+        `submitHeartbeatJob for oracle ${oracleAddress} is no longer active. Removing job.`
+      )
+      job.remove()
```

Comment:
`submitHeartbeatJob` is not repeatable as well, therefore I would recommend that same as above.